osmomath/binary_search.go

```diff
@@ -107,7 +107,7 @@ func (e ErrTolerance) CompareBigDec(expected BigDec, actual BigDec) int {
 	comparisonSign := 0
 	if expected.GT(actual) {
 		comparisonSign = 1
-	} else {
+	} else if expected.LT(actual) {
```

Comment:
Note: I found out that logic here was broken for geometric tolerance

Reply:
Ah your right